src/containers/VoterHistory/index.tsx

```diff
@@ -81,44 +83,46 @@ function VoterHistory(props: Props) {
   }, [historyData, hasParticipations])
 
   useEffect(() => {
-    if (historyData.data && historyData.data.polls) {
-      getPollsData(historyData.data.polls).then(result => {
-        const polls = result.filter(Boolean)
-        setPolls([...polls])
-        Promise.all(
-          polls.map(poll => {
-            return getPollData(poll, pollsBalances).then(data => {
-              return { ...poll, plurality: setPlurality(data) }
-            })
-          }),
-        ).then(pollsWithPluralityAndParticipation => {
-          setPolls(pollsWithPluralityAndParticipation)
-        })
+    if (!historyData || !historyData.data) return
```

Comment:
I think we could make a little performance improvement by adding another condition here, `!Object.keys(pollsBalances).length`.
Since it's initialised as `{}` it will make run this useEffect twice, one without data and the with data. It's being use in line 99

The rest looks great!